packages/lib/delegationCredential/server.ts

```diff
@@ -92,6 +94,14 @@ const _buildCommonUserCredential = ({
     delegatedTo: delegationCredential.serviceAccountKey
       ? {
           serviceAccountKey: delegationCredential.serviceAccountKey,
+          id: delegationCredential.id,
+          ...(delegationCredential.workspacePlatform.slug === "google"
+            ? {
+                key: delegationCredential.accessTokens?.find((token) => token.userId === user.id)?.key,
+              }
+            : {
+                key: delegationCredential.accessTokens?.[0]?.key,
+              }),
```

Comment:
Can we abstract it out better? This is too verbose based on workspace platform checks.

How about we use googleCalendarMetadata and office365CalendarMetadata? 

We could set a property like 
`delegationCredentialAccessTokenPerUser`:true for Google and assume false by default.